### ISSUE TITLE
[ENH] Randomness handling in MiniRocketMultivariate

### ISSUE THREAD

**Issue by ltsaprounis** [alan-turing-institute/sktime#2568]
**Is your feature request related to a problem? Please describe.**
The current implementation is controlling the randomness using `numpy.seed` e.g. [line 140](https://github.com/alan-turing-institute/sktime/blob/2ab70b78caf30f3a5fa1ea516ba4179dd44ba632/sktime/transformations/panel/rocket/_minirocket_multivariate.py#L140). This is considered bad practice according to the numpy docs here: 
https://numpy.org/doc/stable/reference/random/generated/numpy.random.seed.html?highlight=seed#numpy.random.seed

**Describe the solution you'd like**
The suggested approach for controlling the random generator is to create a new RandomState instance. Scikit-Learn has a nice function [sklearn.utils.check_random_state](https://scikit-learn.org/stable/modules/generated/sklearn.utils.check_random_state.html) that takes all the different inputs (`int, None, RandomState`) and returns the appropriate RandomState. The solution should replace `np.random.seed(seed)` with `rng = check_random_state(self.random_state)` and then any calls of the type `np.random.choice(...)`, or `np.random.randint(...)` with `rng.choice(...)`, `rng.randint(...)`. Example code:
```python
from sklearn.utils import check_random_state

...

rng = check_random_state(self.random_state)

# replace all calls from np.random to rng
rng.choice(...)
rng.randint(...)
```

**Comment by Ris-Bali** [comment]
I'll work on this .

**Comment by Ris-Bali** [comment]
I have a question ; I cannot replace the random.seed() with check_random_state(self.random_state) as the function is not inside the class so I tried to apply check_random_state on the seed and got the following error. 
![image](https://user-images.githubusercontent.com/81592570/166318128-90e2dbaa-a1d2-44ae-8bf8-2539d892bf68.png)

**Comment by Ris-Bali** [comment]
Okay so I went through the Numba docs according to them Numba doesn't allow you to create an individual randomstate instance .

**Comment by ltsaprounis** [comment]
Ah! sorry about that. Makes sense now why the code was written like that. Can you paste the link from the docs here?

**Comment by Ris-Bali** [comment]
![image](https://user-images.githubusercontent.com/81592570/166443474-678ca2aa-59f5-433c-b17e-48cfa58c6403.png)


Here's the link to the docs 
https://numba.pydata.org/numba-doc/dev/reference/numpysupported.html#modules

**Comment by ltsaprounis** [comment]
OK - so if I understand this correctly "but with an independent internal state: seeding or drawing numbers from one generator won’t affect the other" we're ok with using the seed like that inside numba functions. Great job figuring this one out @Ris-Bali. Closing this!
